directory.cpp

```diff
@@ -9,154 +9,126 @@ directory::directory(string name, directory *parent):ancestor(name,parent)
 
 directory::~directory()
 {
-    for(auto i:subdirectories){
+    for(auto i:FileSystemObjects){
         delete i;
     }
 }
 
-list<ancestor *> directory::getSubdirectories() const
+list<ancestor *> directory::getFileSystemObjects() const
 {
-    return subdirectories;
+    return FileSystemObjects;
 }
 
 directory *directory::getparent()
 {
-    return this->parent;
+    directory* temp=dynamic_cast<directory*>(this->parent);
+    return temp;
 }
 
-bool directory::isDir() const
-{
-    return true;
-}
-
-
 string directory::getname()
 {
     return this->name;
 }
 
 void directory::ls(){
-    for(ancestor* i:subdirectories){
+    for(ancestor* i:FileSystemObjects){
         cout<<i->getname()<<endl;
     }
 }
 
 bool directory::hasDirs()
 {
-    if(this->getSubdirectories().empty()){
+    if(this->getFileSystemObjects().empty()){
         return false;
     }
     return true;
```

Comment:
`if(vmi) return false else return true`? Szerintem menni fog ez if nelkul is ;-)